From 26606676d9d88c3f301fb8b3d39941ceff267af5 Mon Sep 17 00:00:00 2001
From: "plai@redhat.com" <plai@redhat.com>
Date: Mon, 13 May 2019 16:27:20 +0200
Subject: rhel: Set host-phys-bits-limit=48 on rhel machine-types

RH-Author: plai@redhat.com
Message-id: <1553276715-26278-3-git-send-email-plai@redhat.com>
Patchwork-id: 85126
O-Subject: [RHEL8.0 qemu-kvm PATCH 2/2] rhel: Set host-phys-bits-limit=48 on rhel machine-types
Bugzilla: 1688915
RH-Acked-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>

From: Eduardo Habkost <ehabkost@redhat.com>

Upstream status: not applicable

Currently we use the host physical address size by default on
VMs.  This was a good default on most cases, but this is not the
case on host CPUs supporting 5-level EPT.  On those cases, we
want VMs to use 4-level EPT by default.

Ensure VMs will use 4-level EPT by default, by limiting physical
address bits to 48.

Not applicable upstream because upstream doesn't set
host-phys-bits=on by default.

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
(cherry picked from commit 01a2ecb4c38fe4a35455ea706e76984ee8d5a769)
Signed-off-by: Paul Lai <plai@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>

Rebase notes (4.0.0):
- update to no compat format

Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/i386/pc.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hw/i386/pc.c b/hw/i386/pc.c
index 03497eed9a..91a4c5833a 100644
--- a/hw/i386/pc.c
+++ b/hw/i386/pc.c
@@ -360,6 +360,11 @@ GlobalProperty pc_rhel_compat[] = {
             .property = "host-phys-bits",
             .value = "on",
         },
+        { /* PC_RHEL_COMPAT */ 
+            .driver = TYPE_X86_CPU,
+            .property = "host-phys-bits-limit",
+            .value = "48",
+        },
         { /* PC_RHEL_COMPAT bz 1508330 */ 
             .driver = "vfio-pci",
             .property = "x-no-geforce-quirks",
-- 
2.20.1

